Mage/src/main/java/mage/abilities/meta/OrTriggeredAbility.java

```diff
@@ -72,9 +72,13 @@ public boolean checkEventType(GameEvent event, Game game) {
     public boolean checkTrigger(GameEvent event, Game game) {
         boolean toRet = false;
         for (TriggeredAbility ability : triggeredAbilities) {
+            for (Effect e : getEffects()) { //Add effects to the sub-abilities so that they can set target pointers
+                ability.addEffect(e);
+            }
             if (ability.checkEventType(event, game) && ability.checkTrigger(event, game)) {
                 toRet = true;
             }
+            ability.getEffects().clear(); //Remove afterwards, ensures that they remain synced even with copying
```

Comment:
Though this feels a bit weird, I think I see what this is trying to do and why it can't be done otherwise

Reply:
That’s code with effects copy must be removed 100%. Conditional abilities from “or” parts must have all real effects. If something put effects in “or” ability then that’s wrong logic/design.

Reply:
Any triggered ability that has `SetTargetPointer` needs this code to function correctly, as otherwise that inner ability has no access to the effects it's trying to set the target pointers of.

Reply:
It’s weird. Whole OrTriggeredAbility is big workaround with potentially buggy code (example: if something clear/reset data inside triggered ability or init something then it will be broken with that “container”). Must be reworked/removed someday to simple ability with multiple conditionals.

So new part for temporary add/remove is “fine” here — it’s a fix for existing logic.

![IMG_0705](https://github.com/user-attachments/assets/6d8dbf38-9bcb-48c4-b256-6b0fb7ef2617)

Reply:
Only 53 usages of OrTriggeredAbility. How many of them need target pointers set in order to function? Seems like the sort of thing that could be excluded from scope, with those few usages getting cleaner custom implementations